### PR TITLE
Feat-KG: Integrate Knowledge Graphs (KGs) into the Datasets Tab

### LINKED ISSUE #31: FEATURE: Add Knowledge Graphs (KGs) to the Datasets Tab
**Is your feature request related to a problem? Please describe.**  
We want to extend the functionality of our datasets tab to include Knowledge Graphs (KGs), similar to how SBML models are currently handled. This enhancement would enable users to load open-source KGs and perform tasks such as link prediction and time-series forecasts using various algorithms.

**Describe the solution you'd like.**

The proposed steps are as follows:

- [ ] Convert the current **Data** section into a dropdown menu that allows users to toggle between **SBML models** and **Knowledge Graphs**.
- [ ] Create a new **KGs** page with a table similar to the one used for SBML models. The table should include:
  - A column to view the abstract of the KG
  - Title of the KG
  - Number of nodes
  - Number of edges
- [ ] Develop a **KG dataloader**, inspired by earlier versions of this repo, such as the [Cora KG example](https://github.com/VirtualPatientEngine/VPELeaderboard/tree/v1.0.0/app/data).
- [ ] Initially support **[BioBridge](https://github.com/RyanWangZf/BioBridge)**. This KG is a subset of [PrimeKG](https://github.com/mims-harvard/PrimeKG)**, and includes enriched and embedded multi-modal features. We can build upon the existing [PrimeKG loader tutorial notebook](https://virtualpatientengine.github.io/AIAgents4Pharma/notebooks/talk2knowledgegraphs/tutorial_primekg_loader/) and [BioBridge-PrimeKG Loader](https://virtualpatientengine.github.io/AIAgents4Pharma/notebooks/talk2knowledgegraphs/tutorial_biobridge_primekg_loader/) created by Ahmad and develop one for our repo using the Lightning framework.

_Expected Output_

- [ ] A reusable **dataloader** for loading Knowledge Graphs in PyG format.
- [ ] A **Jupyter notebook** demonstrating how to load and use a KG with the dataloader in the Lightning framework
- [ ] Updates to the **README** and **GitHub Pages website** to include documentation and examples for KGs

### PR DESCRIPTION
**Pull request by Rakesh-Seenu**
# For authors
image
image

## Description
This PR implements Knowledge Graph (KG) support in the Datasets tab, addressing [Issue #31](https://github.com/VirtualPatientEngine/VPELeaderboard/issues/31). Key modifications include:

The main changes include:

- User Interface Update:
The original “Data” section is replaced by a menu that users can choose between viewing SBML models and Knowledge Graphs. This provides a cleaner, more intuitive interface for dataset selection.

- New Knowledge Graphs Page:
A dedicated page has been added for Knowledge Graphs. It displays a table listing all available KGs with important metadata:
  - Title: The name of the Knowledge Graph.
  - Abstract: A brief summary or description of the KG, viewable via a dedicated column.
  - Number of Nodes and Edges: These graph statistics help users understand the size and complexity of each KG at a glance.

- Reusable Knowledge Graph Data Loader:
Inspired by earlier examples such as the Cora dataset loader and PrimeKG tutorials, a new dataloader was implemented. This loader reads KGs in PyTorch Geometric (PyG) format, making it easy to feed graph data into graph neural network models built with the Lightning framework.

- Jupyter Notebook Example:
To help users quickly get started, a notebook demonstrates how to load a KG using the new dataloader and prepare it for use with PyTorch Lightning models. 

- Documentation Updates:
The README and project website now include detailed instructions on working with Knowledge Graphs, describing the new UI components, dataloader usage, and supported datasets like BioBridge.

## Dependencies

- torch-geometric (PyG)
- pytorch-lightning
- networkx
- pandas, numpy (for data preprocessing)
- BioBridge KG (included in data loader)

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests you conducted to verify your changes. These may involve creating new test scripts or updating existing ones.
- [x] Added new test(s) in the ```tests``` folder
- [x] Added new function(s) to an existing test(s) (e.g.: ```tests/testX.py```)
- [ ] No new tests added (Please explain the rationale in this case)

## Checklist
- [x] My code follows the style guidelines mentioned in the Code/DevOps guides
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. MkDocs)
- [ ] My changes generate no new warnings
- [x] I have added or updated tests (in the ```tests``` folder) that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?  
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

### CODE REVIEW

**Comment by Rakesh-Seenu** [comment]
@gurdeep330 Please let me know any changes are required.

**Review by gurdeep330 — CHANGES_REQUESTED** [review]
Thanks for updating the PR @Rakesh-Seenu !
I have a few minor comments, but other than that, it looks good.

File: tests/app/data/test_kg_report.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
+# """
```
[the remark this answers is not in the document]
Comment (Rakesh-Seenu) [reply tests/app/data/test_kg_report.py:1]: No, we don't need that commented code, I will remove it

File: docs/notebooks/KG_dataloader.ipynb [hunk 2]
```diff
@@ -0,0 +1,1264 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "4b467283",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initial\n",
```
[the remark this answers is not in the document]
Comment (Rakesh-Seenu) [reply docs/notebooks/KG_dataloader.ipynb:70]: No, we don't need it

File: tests/vpeleaderboard/data/sbml/test_sys_bio_model.py [hunk 3]
```diff
@@ -4,17 +4,17 @@
...
     MockModel,
     temp_folder_fixture,
     valid_sbml_folder
 )
 
-MODEL_PATH = "vpeleaderboard/data/models/BIOMD0000000064_url.xml"
+MODEL_PATH = "vpeleaderboard/data/src/sbml/models/BIOMD0000000064_url.xml"
 
-class TestSysBioModel(SysBioModel):
+class _ConcreteSysBioModelForTesting(SysBioModel):
```
[the remark this answers is not in the document]
Comment (Rakesh-Seenu) [reply tests/vpeleaderboard/data/sbml/test_sys_bio_model.py:17]: I’ll remove the class entirely since it’s not being called anywhere in the tests. Let me know if you think we should keep it for potential future tests.
